package org.osgi.test.assertj.dto.service.http.runtime.dto;

import org.osgi.service.http.runtime.dto.FailedResourceDTO;

/**
 * Abstract base class for {@link FailedResourceDTO} specific assertions - Generated by CustomAssertionGenerator.
 */
@javax.annotation.Generated(value="assertj-assertions-generator")
public abstract class AbstractFailedResourceDTOAssert<S extends AbstractFailedResourceDTOAssert<S, A>, A extends FailedResourceDTO> extends AbstractResourceDTOAssert<S, A> {

  /**
   * Creates a new <code>{@link AbstractFailedResourceDTOAssert}</code> to make assertions on actual FailedResourceDTO.
   * @param actual the FailedResourceDTO we want to make assertions on.
   */
  protected AbstractFailedResourceDTOAssert(A actual, Class<S> selfType) {
    super(actual, selfType);
  }

  /**
   * Verifies that the actual FailedResourceDTO's failureReason is equal to the given one.
   * @param failureReason the given failureReason to compare the actual FailedResourceDTO's failureReason to.
   * @return this assertion object.
   * @throws AssertionError - if the actual FailedResourceDTO's failureReason is not equal to the given one.
   */
  public S hasFailureReason(int failureReason) {
    // check that actual FailedResourceDTO we want to make assertions on is not null.
    isNotNull();

    // overrides the default error message with a more explicit one
    String assertjErrorMessage = "\nExpecting failureReason of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";

    // check
    int actualFailureReason = actual.failureReason;
    if (actualFailureReason != failureReason) {
      failWithMessage(assertjErrorMessage, actual, failureReason, actualFailureReason);
    }

    // return the current assertion for method chaining
    return myself;
  }

}
